test/instrumentation/modules/mongodb-core.js

```diff
@@ -36,50 +36,17 @@ test('instrument simple command', function (t) {
     })
 
     if (semver.lt(version, '2.0.0')) {
-      // mongodb-core v1.x will sometimes perform two `ismaster` queries
```

Comment:
This isn't related to Node.js v7 and below, but to mongodb-core v1.x. I just tried running this branch locally with the latest v1.x and it failed. What's weird though is that the TAV tests on Jenkins didn't fail 🤔 Maybe we were just "lucky" that the driver didn't decide to perform an `admin.$cmd.ismaster` check that time around - I'll try to trigger them again

Reply:
The tests just ran a 2nd time and all passed... I'm really not sure why because they all fail on my machine??!?

What happens if you run them on yours?

Reply:
Yeah, 1.x is what I was testing against. I ran probably about a hundred times locally and _never_ got that extra command, except on 7.x. 🤔 

Maybe you're running against an old mongodb? I did most of my testing with `.ci/scripts/test.sh $NODE_VERSION mongodb-core false`, and ran probably 20+ times on each major from 6 up, and 7.x was the only one that encountered the extra command for me. Maybe against a different version of mongo it has different behaviour? 🤔

Reply:
I'm running against whatever Docker gives me... maybe it's a cache thing because we don't specify a version in `docker-compose.yml`:

https://github.com/elastic/apm-agent-nodejs/blob/66765c1e473d5274e9ee8bd817ed1d4ed6faa9d8/test/docker-compose.yml#L20-L30

If I run `docker image inspect mongo`, I get the following ID:

```
sha256:f7adfc4dbcf55493944234d9c9aea9c5b8605a1ccfe77c1ddbcbd016bfff531b
```

This is how I got the error to appear last week:

1. First I check out your branch:
    ```sh
    git checkout -b Qard-fix-mongo-tav master
    git pull https://github.com/Qard/apm-agent-nodejs.git fix-mongo-tav
    ```
1. Then I install the latest version 1.x of `mongodb-core`:
    ```sh
    npm install mongodb-core@1
    ```
1. Then I ran the mongodb-core test:
    ```sh
    node test/instrumentation/modules/mongodb-core.js
    ```

**But now when I just tried to do the same thing again, it all worked fine?!?** I guess this is either random or based on some state which is stored somewhere.

Reply:
I got the extra events _sometimes_ when running directly on the machine with `node test/instrumentation/modules/mongodb-core.js`, but when I ran _everything_ in containers using `.ci/scripts/test.sh $NODE_VERSION mongodb-core false` I consistently never got the extra events.

Reply:
Confirmed: it's a macOS thing. I restored the logic to support macOS, but only when `platform` is `darwin`.

Reply:
I think we're talking about two different things. I'm talking about how the first element in the `data.spans` array sometimes has the name `admin.$cmd.ismaster`.

Two days ago I ran this locally and printed out the content of `data.spans` and this was the result:

```js
[
  {
    id: '2610464e6275dc00',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'admin.$cmd.ismaster',
    type: 'db.mongodb.query',
    timestamp: 1568365847525436,
    duration: 6.441,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: '46025110fc077b21',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'system.$cmd.ismaster',
    type: 'db.mongodb.query',
    timestamp: 1568365847533065,
    duration: 1.876,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: '4ea3d486f5d8277d',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'elasticapm.test.insert',
    type: 'db.mongodb.query',
    timestamp: 1568365847535619,
    duration: 2.653,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: '7c755ea676b32cc6',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'elasticapm.$cmd.command',
    type: 'db.mongodb.query',
    timestamp: 1568365847535960,
    duration: 2.243,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: 'cf6c13c7525b5c03',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'elasticapm.test.update',
    type: 'db.mongodb.query',
    timestamp: 1568365847538479,
    duration: 2.635,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: '62e82630c30006f0',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'elasticapm.$cmd.command',
    type: 'db.mongodb.query',
    timestamp: 1568365847538856,
    duration: 2.219,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: '2a16d561a0d717f3',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'elasticapm.test.remove',
    type: 'db.mongodb.query',
    timestamp: 1568365847541293,
    duration: 2.448,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: '31f0b14e14f5f635',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'elasticapm.$cmd.command',
    type: 'db.mongodb.query',
    timestamp: 1568365847541664,
    duration: 2.045,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: 'd0f1340fff2894e1',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'elasticapm.test.find',
    type: 'db.mongodb.query',
    timestamp: 1568365847544120,
    duration: 3.239,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  },
  {
    id: 'ebf5cb38a5821870',
    transaction_id: 'ca9f271dec8b5a65',
    parent_id: 'ca9f271dec8b5a65',
    trace_id: '8f49747d568d6eb60d9e2197d0b98efd',
    name: 'system.$cmd.ismaster',
    type: 'db.mongodb.query',
    timestamp: 1568365847547662,
    duration: 1.829,
    context: undefined,
    stacktrace: [ [Object], [Object], [Object], [Object], [Object] ],
    sync: false
  }
]
```

Reply:
Yes, that's what I'm talking about. Sometimes there's an extra `admin.$cmd.ismaster` span at the start, sometimes there's an extra `system.$cmd.ismaster` command at the start, but this _appears_ to actually only be the case on macOS.

Reply:
Doh! I somehow only noticed your first `darwin` check but didn't see that you were manipulating the `groups` array afterwards. Sorry - all is good then